[caffe2] exposes Net cancellation through pybind state

dahsh opened the pull request:
Summary: * To invoke `cancel` from the net instance in Python, we expose it through bybind state.

Differential Revision: D23249660

d4l3k commented:
This seems fine to me since it's just exposing more of the C++ API to python. @dzhulgakov, @ilia-cher could one of you sanity check this?

dahsh committed:
[caffe2] exposes Net cancellation through pybind state (#44043)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/44043

To invoke `cancel` from the net instance in Python, we expose it through pybind state.

Reviewed By: dzhulgakov

Differential Revision: D23249660

fbshipit-source-id: 83e8651e27c236ef7da9daa44459fbf2655a0b3c
